src/modules/peek/Peek.UI/PeekXAML/Views/TitleBar.xaml.cs

```diff
@@ -1,357 +0,0 @@
-// Copyright (c) Microsoft Corporation
-// The Microsoft Corporation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System;
-using System.Collections.Generic;
-using System.Globalization;
-using CommunityToolkit.Mvvm.ComponentModel;
-using CommunityToolkit.Mvvm.Input;
-using ManagedCommon;
-using Microsoft.PowerToys.Telemetry;
-using Microsoft.UI;
-using Microsoft.UI.Windowing;
-using Microsoft.UI.Xaml;
-using Microsoft.UI.Xaml.Controls;
-using Peek.Common.Helpers;
-using Peek.Common.Models;
-using Peek.UI.Extensions;
-using Peek.UI.Helpers;
-using Peek.UI.Telemetry.Events;
-using Windows.Graphics;
-using Windows.Storage;
-using Windows.System;
-using WinUIEx;
-
-namespace Peek.UI.Views
-{
-    [INotifyPropertyChanged]
-    public sealed partial class TitleBar : UserControl
-    {
-        public static readonly DependencyProperty ItemProperty =
```

Comment:
You should add an additional dependency property here with the needed value. As you can see in the registration, you can define a callback when the property gets changed and react on it.

Reply:
For example, you could define dependency property for the following properties:

- WindowActivationState (to get the active/disable state of the window to change the title in the titlebar)
- AppWindow (to access the properties that were previously null)

Reply:
Okay. This is where it gets tricky for me. Don't know if you have access/rights to commit to my branch, but if you do, feel free to add changes :)